Source/player.cpp

```diff
@@ -3214,12 +3207,14 @@ void StripTopGold(Player &player)
 			if (item._ivalue > MaxGold) {
 				int val = item._ivalue - MaxGold;
 				item._ivalue = MaxGold;
-				SetPlrHandItem(player.HoldItem, 0);
+				InitializeItem(player.HoldItem, IDI_GOLD);
 				SetGoldSeed(player, player.HoldItem);
 				player.HoldItem._ivalue = val;
 				SetPlrHandGoldCurs(player.HoldItem);
-				if (!GoldAutoPlace(player))
-					DeadItem(player, &player.HoldItem, { 0, 0 });
+				if (!GoldAutoPlace(player)) {
+					DeadItem(player, player.HoldItem, { 0, 0 });
+					player.HoldItem._itype == ItemType::None;

```

Comment:
This gets overwritten by line 3222 so not really needed. But eventually we should probably not use `player.HoldItem` as the temp item.

Reply:
Not entirely sure why I didn't remove the use of `player.HoldItem` here, `DeadItem` no longer uses holditem as a temporary so I can create a new item instance instead

Reply:
Because the auto-pickup gold code used holditem as a temporary as well -_-. I'll include that in the next push